src/modules/Modal/Modal.js

```diff
@@ -1,49 +1,67 @@
-import React, { Component, PropTypes } from 'react'
-import classNames from 'classnames'
+import React, { PropTypes } from 'react'
+import cx from 'classnames'
 import META from '../../utils/Meta'
 import ModalContent from './ModalContent'
 import ModalHeader from './ModalHeader'
 import ModalFooter from './ModalFooter'
+import {
+  getUnhandledProps,
+  useKeyOnly,
+} from '../../utils/propUtils'
 
-export default class Modal extends Component {
-  static propTypes = {
-    children: PropTypes.any,
-    className: PropTypes.string,
-    settings: PropTypes.object,
-  }
-
-  state = { isShown: false }
-
-  static Content = ModalContent
-  static Header = ModalHeader
-  static Footer = ModalFooter
-
-  showModal = () => {
-    this.setState({ isShown: true })
-  }
-
-  hideModal = () => {
-    this.setState({ isShown: false })
-  }
-
-  static _meta = {
-    library: META.library.semanticUI,
-    name: 'Modal',
-    type: META.type.module,
-  }
-
-  render() {
-    const classes = classNames(
-      'ui',
-      this.props.className,
-      'modal',
-      { 'transition visible active': this.state.isShown },
-    )
-
-    return (
-      <div {...this.props} className={classes}>
-        {this.props.children}
-      </div>
-    )
-  }
+function Modal(props) {
+  const {
+    children, className,
+    active, basic, size, long,
+  } = props
+
+  const classes = cx('ui modal',
+    useKeyOnly(active, 'transition visible active'),
+    useKeyOnly(basic, 'basic'),
+    useKeyOnly(long, 'long'),
+    size,
+    className,
+  )
+
+  const rest = getUnhandledProps(Modal, props)
+
+  return (
+    <div className={classes} {...rest} >
+      {children}
+    </div>
+  )
+}
+
+Modal._meta = {
+  name: 'Modal',
+  type: META.type.module,
+  props: {
+    size: ['fullscreen', 'large', 'small'],
+  },
+}
+
+Modal.propTypes = {
+  /** Primary content of the modal. Consider using ModalHeader, ModalContent or ModalFooter here */
```

Comment:
Also, let's use `PropTypes.node` which per the [React docs](https://facebook.github.io/react/docs/reusable-components.html) is "anything that can be rendered"